### PR TITLE
Refactor/clang tidy missing braces

### PR DESCRIPTION
**Pull request by SteveBronder**
## Summary
I ran clang-tidy over the repo and found a bunch of places we don't use braces around if, for, etc. It would be nice to have that for readability. This PR adds braces around a lot of the conditional statements etc. that were missing them in the math library!

## Tests

No new tests

## Checklist

- [x] Math issue #1308 

- [x] Copyright holder: Steve Bronder

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: make/compiler_flags [hunk 1]
```diff
@@ -13,7 +13,7 @@ ifeq (default,$(origin CXX))
     CXX := clang++
   endif
   ifeq ($(OS),Linux)
-    CXX := g++
+    CXX := clang++
```
Comment (SteveBronder) [inline make/compiler_flags]: A to-do here, clang-tidy uses clang, so I'm not sure how to go around having g++ here though I'd like to keep it.

**Comment by SteveBronder** [comment]
@syclik who should review this?

File: make/compiler_flags (hunk 1 above)
Comment (rok-cesnovar) [reply make/compiler_flags]: Do we really need to change this here? If we make this a part of the Jenkins process, we can set CXX = clang++ there before running make clang-tidy. Like we set flags for MPI or OPENCL before running those tests.

Comment (SteveBronder) [reply make/compiler_flags]: Yeah idt we need it I'll remove it

**Comment by SteveBronder** [comment]
Gonna close this until we bring in clang-tidy makefile stuff

**Pull request closed without merging**